enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,417 @@
+
+* Assumption: customers siteconfig consists 2 extra manifests paths which are in the user Git repository
+
+  ```yaml
+    - cluster:
+      extraManifestSearchPaths: 
+        - sno-extra-manifest/
+        - custom-manifests/
+      extraManifests:
+        filter:			
+    ```
+
+  
+* Those path conatins below files:
+
+  ```bash
+  extra manifests in the sno-extra-manifest
+
+  - A.yaml
+  - B.yaml
+  - C.yaml 
+  ```
+
+  ```bash
+  extra manifests in the custom-manifests
+
+  - C.yaml
+  - D.yaml
+  - E.yaml 
+  ```
+
+<table>
+<tr>
+<th>
+Scenario
+</th>
+<th>
+Logical Flow
+</th>
+<th>
+Output CR list
+</th>
+</tr>
+
+<tr>
+<td>
+Include all files from the path
+<pre>
+ yaml
+ [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/			
+  ]
+</pre>
+</td>
+
+
+
+<td>
+  <li>When <b>extraManifestSearchPaths</b> is defined all the CRs will be added in site generation</li>
+ <li>In this example, all the CRs from user git will be added</li>
+</td>
+
+<td>
+  [A.yaml B.yaml C.yaml D.yaml E.yaml] <-- C.yaml picked from <b>custom-manifests</b>
+</td>
+</tr>
+
+<tr>
+<td>
+Siteconfig contains both parameter
+<pre>
+ yaml
+ [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/
+    extraManifestPath: custom-manifests/
+  ]
+</pre>
+</td>
+
+
+<td>
+  <li>Proposed 4.14 behavior will be applied. SiteGen will discard extraManifestPath, it will only add files from <b>extraManifestSearchPaths</b></li>
+
+</td>
+
+<td>
+ [A.yaml B.yaml C.yaml D.yaml E.yaml] <-- C.yaml picked from <b>custom-manifests</b> of extraManifestSearchPaths path
+</td>
+
+</tr>
+<td>
+Exclude some files
+<pre>
+ yaml
+ [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - C.yaml
+          - D.yaml
+  ]
+</pre>
+</td>
+
+
+<td>
+  <li>The behavior of filter will not change</li>
+  <li>if the default is <i>exclude</i>, only listed *.yaml files will be included in CR generation</li>
+</td>
+
+<td>
+ [C.yaml D.yaml] <--C.yaml picked  <b>custom-manifests</b>
+</td>
+
+</tr>
+
+
+<tr>
+<td>
+Include only mentioned files
+<pre>
+ yaml
+  [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/
+    extraManifests:
+      filter:
+        inclusionDefault: include
+        exclude:
+          - C.yaml
+          - D.yaml
+  ]
+</pre>
+</td>
+
+
+<td>
+  <li>The behavior of filter will not change</li>
+  <li>if the default is <i>include</i>, only listed *.yaml files will be excluded in CR generation</li>
+</td>
+
+<td>
+ [A.yaml B.yaml E.yaml]
+</td>
+
+</table>
+         
+
+### Risks and Mitigations
+
+N/A
+
+## Implementation choices
+
+N/A
+
+### API Extensions
+
+New parameter will be introduced named `extraManifestSearchPaths` which will accept a `list[]`
```

Comment:
So if the list's length is greater than 0...it's implied that the user wants version independence?

Reply:
Not necessarily. For example:

```
extraManifestSearchPaths:
      - testManifest/extra-manifest
      - customManifests
```      

user can put builtin CRs coming from the release specific ztp-container in path `testManifest/extra-manifest` and their custom CRs in `customManifests`, and if one/multiple CRs found in both path, the later will get precedence.